project-base/storefront/urql/cache/updates.ts

```diff
@@ -93,6 +93,9 @@ export const cacheUpdates: UpdatesConfig = {
         RemoveProductList(_result, args: RemoveProductListMutationVariablesApi, cache) {
             manuallyRemoveProductListQuery(cache, args.input);
         },
+        ChangePaymentInOrder(_result, _args, cache) {
+            invalidateFields(cache, ['order']);
+        },
```

Comment:
It is possible to do it in a more optimal way by taking the new payment and updating the order query with it. You save a query. Not sure how helpful that is in this case. I will leave it for you to decide.

Reply:
Hmm, you mean something like manuallyRemoveProductListQuery, but with an order?. Maybe this will be a cleaner solution, but I think it's a bit overkill because this is only for the order detail with gopay and the optimization will be some small numbers.
If you have other points about it, I am open to discussing 🙂